### PR TITLE
Modernbert Release Fixes

### PR DESCRIPTION
**Pull request by warner-benjamin**
This PR resolves two issues with ModernBERT so it will be working for release in just under an hour.

First, ModernBERT has a final head layer `ModernBertPredictionHead` with a pretrained Linear layer and LayerNorm . In downstream heads, these two layers are loaded to `ModernBertPoolingHead`. I readded `ModernBertPoolingHead` to `ModernBertForTokenClassification` so TokenClassification models will load the pretrained layer weights.

Second, #35235 appears to have broken the SDPA/Eager paths by changing how the RoPE layer ModernBERT inherited from works. In the interest of expediency, I unmoduarized it so we can have a working ModernBERT implementation on all paths.

cc @ArthurZucker @tomaarsen

### CODE REVIEW

**Review by tomaarsen — APPROVED** [review]
The rotary embedding solution is a temporary fix - I'll introduce a modular-approved one soon, but we have a release to hit.

I didn't realise the ForTokenClassification model also requires the dense -> act -> norm -> drop, I thought it was only drop like with BERT. My apologies, thanks for the quick fix.

File: src/transformers/models/modernbert/modular_modernbert.py [hunk 1]
```diff
@@ -1247,13 +1271,16 @@ def __init__(self, config: ModernBertConfig):
...
+    def forward(
+        self, hidden_states: torch.Tensor, attention_mask: torch.Tensor, apply_pooling: bool = True
+    ) -> torch.Tensor:
+        if apply_pooling:
+            if self.config.classifier_pooling == "cls":
+                hidden_states = hidden_states[:, 0]
+            elif self.config.classifier_pooling == "mean":
+                hidden_states = (hidden_states * attention_mask.unsqueeze(-1)).sum(dim=1) / attention_mask.sum(
+                    dim=1, keepdim=True
+                )
```
Comment (ArthurZucker) [inline src/transformers/models/modernbert/modular_modernbert.py:1283]: that is even more code pathes. At this point are are the hidden states passed to the model correct? Like sequence classification passes.
Depending on the task, the hidden states should be approprioatly passed to the layer

File: src/transformers/models/modernbert/modular_modernbert.py [hunk 2]
```diff
@@ -1422,7 +1449,7 @@ def forward(
         )
         last_hidden_state = outputs[0]
 
-        last_hidden_state = self.drop(last_hidden_state)
+        last_hidden_state = self.head(last_hidden_state, attention_mask, apply_pooling=False)
```
Comment (ArthurZucker) [inline src/transformers/models/modernbert/modular_modernbert.py:1452]: if it is always false, why do we have an argument for it?

Comment (warner-benjamin) [reply src/transformers/models/modernbert/modular_modernbert.py:1452]: The pooling layer needs to pools for heads which need it, like `ModernBertForSequenceClassification`, and not pool for heads which don't, like `ModernBertForTokenClassification`.

I can swap `ModernBertPoolingHead` out for the `ModernBertPredictionHead` in the `ModernBertForTokenClassification`, if you want.

I thought this way would be less confusing and easier to follow when/if future ForTask heads get added

File: src/transformers/models/modernbert/modular_modernbert.py (hunk 1 above)
Comment (tomaarsen) [reply src/transformers/models/modernbert/modular_modernbert.py]: Resolved in f2be82f2b777e901add429c4892bb0cdfba5a1f5

**Review by ArthurZucker — APPROVED** [review]
Thanks 🤗

**Comment by sileod** [comment]
Hi, thanks, is there a plan to support multiplechoice ?

**Comment by stefan-it** [comment]
Have you already road-tested the token classification implementation? E.g. using the ModernBERT Large model only yields to ~91% F-Score on CoNLL-2003 (Test set, I tried some batch size and learning rate configurations), in two different frameworks (Transformers and Flair). I am not sure if ModernBERT needs really special hyper-parmaters or if there are fixes needed in the architecture :thinking:

**Comment by KoichiYasuoka** [comment]
Hi @stefan-it I've tried to test `ModernBertForTokenClassification` with [ModernBERT-base](https://huggingface.co/answerdotai/ModernBERT-base) and [UD_English-EWT](https://github.com/UniversalDependencies/UD_English-EWT) and got 97% F1-Score on UPOS (see [my report](https://qiita.com/KoichiYasuoka/items/d1fbb04804079a4e495e) written in Japanese). During my test I found that `ModernBertForTokenClassification` lacked `get_input_embeddings()` and made PR #35422. And I seem the tokenizer of ModernBERT-base is slightly difficult to handle POS-tagging (than BERT's one) and `"classifier_dropout": 0.0` seems too small.
